content/docs/pipelines/sdk/python-based-visualizations.md

```diff
@@ -0,0 +1,177 @@
++++
+title = "Python Based Visualizations"
+description = "Predefined and arbitrary visualizations of pipeline outputs"
+weight = 80
++++
+
+This page describes python based visualizations, how to create them, and how to
+use them to visualize results within the Pipelines UI.
+
+## Introduction
+
+Python based visualizations are a new method to visualize results within the
+Pipelines UI. This new method of visualizing results is done through the usage
+of [nbcovert](https://github.com/jupyter/nbconvert), the tool used by Jupyter
+Notebook that generates outputs. Alongside the usage of nbconvert, results of a
+pipeline can now be visualized without a component being included within the
+pipeline itself. The process of visualizing results are now decoupled from a
+pipeline.
+
+Python based visualizations provide two categories of visualizations. The first
+being **predefined visualizations**. These visualizations are curated by the
+Kubeflow Pipelines Team and serve as a way for users to easily and quickly
+generate powerful visualizations. However, these are not limited to just the
+Pipelines Team and this will be discussed further below in
+[Why Predefined Visualizations are Important](/docs/pipelines/sdk/
+python-based-visualizations/#why-predefined-visualizations-are-important).
+The second category is **arbitrary visualizations**. Arbitrary visualizations
+allow for user provided python visualization code to be used to generate
+visualizations. These visualizations allow for rapid development,
+experimentation, and customizability when visualizing results.
+
+## Why Predefined Visualizations are Important
+
+Predefined visualizations provide major benefits for the Pipelines Team and
+users of Pipelines. The Pipelines Team is able to develop predefined
+visualizations to provide users of Pipelines with powerful, extensible, and
+standardized visualizations that work out of the box. For users of Pipelines,
+they are able to utilize existing predefined visualization and build their own.
+For some users, such as those who use Pipelines with a team or those who
+frequently generate complex and specialized visualizations, predefined
+visualizations can be leveraged to reduce development time and provide
+standardizations of visualized results. Details about how to create predefined
+visualizations can be found [below](/docs/pipelines/sdk/
+python-based-visualizations/#how-to-create-predefined-visualizations).
+
+## How to Create Predefined Visualizations
+
+1. Determine what the new visualization will be.
+2. Add a new type for the visualization within the.
+[visualization.proto](https://github.com/kubeflow/pipelines/blob
+/master/backend/api/visualization.proto#L78) file.
+    * The name of the visualization should be in screaming snake case
+    (i.e. `VISUALIZATION_NAME`).
```

Comment:
Spell out "i.e." as "that is," - see the style guide. Many readers confuse "i.e." and "e.g.", and Latin abbreviations are confusing for many people whose first language is not English or similar.